docs/oauth.md

```diff
@@ -0,0 +1,215 @@
+| CAP_FIXED_PORT_IN_CALLBACK_URL | Use this when your OAuth provider native app(Tableau Desktop) OAuth clients only support fixed callback url | false | - |
+| CAP_SUPPORTS_HTTP_SCHEME_LOOPBACK_REDIRECT_URLS | Use this when your OAuth provider native app(Tableau Desktop) OAuth clients support Loopback callback url. E.g. https://developers.google.com/identity/protocols/oauth2/native-app | false | - |
+| CAP_REQUIRES_PROMPT_CONSENT | Add prompt=consent to the request. | false | - |
+| CAP_REQUIRES_PROMPT_SELECT_ACCOUNT | Add propmt=select_account to the request. More details: https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow | false | - |
+| CAP_SUPPORTS_GET_USERINFO_FROM_ID_TOKEN | Used when your OAuth response contains a JWT style ID_TOKEN that can be parsed out to get actual username. e.g. https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens | false | - |
+
+## Example OAuthConfig file:
+**oauthConfig.xml**
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<pluginOAuthConfig>
+    <!-- dbclass must correspond to the class registered in manifest.xml --> 
+    <dbclass>snowflake_oauth</dbclass>
+    <!-- Subsitute these with your OAuth Client Id and Client Secret -->
+    <clientIdDesktop>[your_client_id]</clientIdDesktop>
+    <clientSecretDesktop>[your_client_secret]</clientSecretDesktop>
+    <-- Desktop redirectUri, subsitute for your own registered desktop redirectUri --> 
+    <redirectUrisDesktop>http://localhost:55555/Callback</redirectUrisDesktop>
+    <!-- authUri and tokenUri only contains the path since it has CAP_SUPPORTS_CUSTOM_DOMAIN on, so the final oauth endpoint will be 
+    your input instanceUrl + authUri/tokenUri -->
+    <authUri>/oauth/authorize</authUri>
+    <tokenUri>/oauth/token-request</tokenUri>
+    <!-- Used to prevent malicious instanceUrl, your instanceUrl must match this regular expression or OAuth flow will abort -->
+    <instanceUrlValidationRegex>^https:\/\/(.+\.)?(snowflakecomputing\.(com|us|cn|de))(.*)</instanceUrlValidationRegex>
+    <!-- Snowflake need refresh_token scope to issue refresh tokens -->
+    <scopes>refresh_token</scopes>
+    <!-- Snowflake supports PKCE, Cutom domain, fixed callback url -->
+    <capabilities>
+        <entry>
+            <key>CAP_PKCE_REQUIRES_CODE_CHALLENGE_METHOD</key>
+            <value>true</value>
+        </entry>
+        <entry>
+            <key>CAP_SUPPORTS_CUSTOM_DOMAIN</key>
+            <value>true</value>
+        </entry>
+        <entry>
+            <key>CAP_REQUIRE_PKCE</key>
+            <value>true</value>
+        </entry>
+        <entry>
+            <key>CAP_FIXED_PORT_IN_CALLBACK_URL</key>
+            <value>true</value>
+        </entry>
+        <entry>
+            <key>CAP_PKCE_REQUIRES_CODE_CHALLENGE_METHOD</key>
+            <value>true</value>
+        </entry>
+    </capabilities>
+    <!-- Map Tableau recognized attr <key> to Snowflake OAuth response attr <value> -->
```

Comment:
Is this snowflake specific, or general?

Reply:
This is general I'll change the comments.